### PR TITLE
Apply once when WatchInterval == 0 

### PR DESCRIPTION
**Pull request by zsuzhengdu**
## Changes

Fix https://github.com/prometheus-operator/prometheus-operator/issues/3061#issuecomment-619812179

## Verification

Manually tested. Working on unit tests.

### CODE REVIEW

**Review by bwplotka — COMMENTED** [review]
Makes sense to me, thanks.

However, I have a small suggestion. What about doing one-off if `WatchInterval time.Duration is 0`? This makes sense logically and will allow us to have consistent API for reloader. 🤗 WDYT?

**Comment by zsuzhengdu** [comment]
@bwplotka  Would you please help to give another review on this PR? Thanks!

**Review by bwplotka — DISMISSED** [review]
Solid. Thanks!

File: pkg/reloader/reloader.go [hunk 1]
```diff
@@ -197,6 +197,11 @@ func (r *Reloader) Watch(ctx context.Context) error {
 		}
 	}
 
+	if r.watchInterval == 0 {
+		// Skip watching the file-system
```
Comment (bwplotka) [inline pkg/reloader/reloader.go:201]:
```suggestion
		// Skip watching the file-system.
```

**Review by bwplotka — APPROVED** [review]

**Pull request merged**